docs/federation.md

```diff
@@ -0,0 +1,126 @@
+## Federation
+
+AMKO has a federator which can federate the AMKO configuration to a set of member clusters. This is done to ensure that a user can initiate a recovery whenever there's some kind of a disaster (without the need of backing up the AMKO configs).
+
+### Federation Set
+Federation works on a set of clusters where AMKO is deployed. We will call this set of clusters as the federation set. One of these clusters in the federation set must be designated as the **leader**. And, all the other clusters are designated as **followers**.
+
+### Responsibilities of a leader and a follower
+1. A **leader** cluster is responsible for distributing the AMKO configuration to all the follower clusters in the federation set. AMKO in the leader cluster is also responsible to create/update GSLB Services on the Avi Controller.
+2. AMKO in a **follower** cluster runs in a no-op mode. The federator in the follower clusters won't federate the objects. And, AMKO in the follower clusters won't sync the GSLB Services in the Avi Controller. It just runs in a passive mode.
+
+Whenever there's a disaster, taking the leader cluster down, the user can then pick any of the follower clusters and mark this AMKO to be the new leader. Though, **note** that at any point in time, the user has to ensure that there's only one leader AMKO.
+
+### Objects to be federated
+Following objects are federated from the leader cluster to the follower:
+1. `GSLBConfig`
+2. `GlobalDeploymentPolicy` or `GDP`
+
+Add/Update/Delete events of the above objects are federated to the follower clusters.
+
+### Flow
+Assuming that there are three sites, and each site has one cluster:
+- Cluster 1 in Site 1
+- Cluster 2 in Site 2
+- Cluster 3 in Site 3
+
+![Alt text](images/amko_fed_initial.png?raw=true "AMKO federation flow")
+
+Each site has an Avi Controller deployed. Site 1's Avi Controller is chosen as the GSLB Leader and all the other sites are GSLB followers. AMKO is deployed on all three sites. For cluster 1 (in site 1), it is marked as the leader. And, for clusters 2 and 3, it is marked as follower. User adds/updates the `GSLBConfig` and `GDP` objects in cluster 1, AMKO's federator in cluster 1 federates the changes to these objects to all the follower clusters.
+
+### AMKOCluster CRD to control federation
+A CRD called `AMKOCluster` governs the federation. With this CR, users can specify which clusters to federate on, along with a few other parameters.
```

Comment:
This line is not required: With this CR, users can specify which clusters to federate on, along with a few other parameters.